dwa_local_planner/include/dwa_local_planner/dwa_planner.h

```diff
@@ -174,6 +174,7 @@ namespace dwa_local_planner {
       double forward_point_distance_;
 
       std::vector<geometry_msgs::PoseStamped> global_plan_;
+      std::vector<geometry_msgs::PoseStamped> actual_global_plan_;
```

Comment:
Maybe original_global_plan_ or something?  "actual" isn't as descriptive.

Reply:
This is already being stored in the local_planner_util_ class.

Reply:
The global plan in the local_planner_util gets pruned.